qubes/storage/lvm.py

```diff
@@ -528,7 +528,7 @@ def import_volume(self, src_volume):
         # HACK: neat trick to speed up testing if you have same physical thin
         # pool assigned to two qubes-pools i.e: qubes_dom0 and test-lvm
         # pylint: disable=line-too-long
-        if isinstance(src_volume.pool, ThinPool) and \
+        if hasattr(src_volume.pool, 'thin_pool') and \
```

Comment:
I know it's tempting, but this may bypass some of the callbacks and cause the operation to fail or copy wrong data (if callbacks setup some extra layer like encryption on top).

Reply:
>  during setup it would immediately max out the pool at 1 GB because I'd put 1 GB random data in the thin volume (otherwise I'd leak the size of the data it keeps).

The assumption here may not be true in all the cases. Not filling free space with random data (or perhaps just encrypted zeroes) indeed leaks info how much space is used, but it may be fine for many use cases. For example you may have a "browsing-work" VM with just firefox - don't leak much when it is known that there is a firefox profile, but you want to protect its content like bookmarks, browsing history, cookies etc.

> Then of course referencing one thin volume from another to remain "thin" doesn't work anymore (I think that's what the "hack" does, isn't it?) because the former may still be encrypted.

Yes, such approach indeed wouldn't work for TemplateVM's root volume of DisposableTemplateVM's private volume. Assuming of course each VM would get a different encryption key.

>  Once again this indicates that encryption per Qubes OS pool is more natural than encryption per Qubes volume.

Yes, but then you loose free disk space sharing between those pools. You need to allocate their sizes statically.

> However if you still want to support the per-volume idea, I guess we'd require `post_volume_import` & `post_volume_create` 

Yes, I think so.